x-pack/plugins/aiops/public/components/explain_log_rate_spikes/explain_log_rate_spikes_analysis.tsx

```diff
@@ -51,6 +52,21 @@ export const ExplainLogRateSpikesAnalysis: FC<ExplainLogRateSpikesAnalysisProps>
   const { services } = useAiOpsKibana();
   const basePath = services.http?.basePath.get() ?? '';
 
+  const [currentAnalysisWindowParameters, setCurrentAnalysisWindowParameters] = useState<
+    WindowParameters | undefined
+  >();
+  const [shouldRerunAnalysis, setShouldRerunAnalysis] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (
+      currentAnalysisWindowParameters !== undefined &&
+      !isEqual(currentAnalysisWindowParameters, windowParameters)
+    ) {
+      console.log('STALE!!'); // remove
+      setShouldRerunAnalysis(true);
+    }
+  }, [currentAnalysisWindowParameters, windowParameters]);
```

Comment:
❓ do you actually need a state and effect for `shouldRerunAnalysis`? 
```suggestion
  const shouldRerunAnalysis = useMemo(() => {
     return currentAnalysisWindowParameters !== undefined &&
      !isEqual(currentAnalysisWindowParameters, windowParameters);
  }, [currentAnalysisWindowParameters, windowParameters]);
```

Reply:
Good suggestion! When I tested this it also seems to fix a bug with the button always staying in `warning` state even if the analysis is rerun.

Reply:
@walterra - that bug you were seeing has been fixed. Moving `shouldRerunAnalysis` into a useMemo makes it so that we never get the stale message since we're now setting current window params to window params when we detect a change and they aren't equal. It's working as expected keeping this value in state.

Reply:
thanks for the explanation @alvarezmelissa87. As you're setting currently window params to window params, is there any other value that indicates it's stale? If not, I find this approach incorrect. You're resettings params programmatically, but the state remains stale.